apps/etterlatte-beregning/src/test/kotlin/BeregningRepositoryImplTest.kt

```diff
@@ -0,0 +1,75 @@
+import io.mockk.mockk
+import io.mockk.mockkClass
+import no.nav.etterlatte.BeregningRepository
+import no.nav.etterlatte.BeregningRepositoryImpl
+import no.nav.etterlatte.DataSourceBuilder
+import no.nav.etterlatte.libs.common.behandling.BehandlingType
+import no.nav.etterlatte.libs.common.grunnlag.Metadata
+import no.nav.etterlatte.libs.common.toJson
+import no.nav.etterlatte.libs.common.vilkaarsvurdering.Vilkaarsvurdering
+import no.nav.etterlatte.model.Beregning
+import no.nav.etterlatte.model.BeregningService
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.testcontainers.containers.PostgreSQLContainer
+import org.testcontainers.junit.jupiter.Container
+import java.time.YearMonth
+import java.util.*
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+internal class BeregningRepositoryImplTest {
+
+    @Container
+    private val postgreSQLContainer = PostgreSQLContainer<Nothing>("postgres:14")
+    private lateinit var beregningRepository: BeregningRepository
+
+    @BeforeAll
+    fun beforeAll() {
+        postgreSQLContainer.start()
+        val ds = DataSourceBuilder(
+            postgreSQLContainer.jdbcUrl,
+            postgreSQLContainer.username,
+            postgreSQLContainer.password
+        ).also { it.migrate() }
+
+        beregningRepository = BeregningRepositoryImpl(ds.dataSource())
+    }
+
+    @AfterAll
+    fun afterAll() {
+        postgreSQLContainer.stop()
+    }
+
+    @Test
+    fun lagreOgHent() {
+        val opplysningsgrunnlag = GrunnlagTestData().hentOpplysningsgrunnlag()
+
+        val beregningResultat = BeregningService(beregningRepository, mockk()).beregnResultat(
+            opplysningsgrunnlag,
+            YearMonth.of(2021, 2),
+            YearMonth.of(2021, 9),
+            mockkClass(Vilkaarsvurdering::class),
+            BehandlingType.FØRSTEGANGSBEHANDLING
+        )
+        val behandlingId = UUID.randomUUID()
+
+        val beregningLagret = Beregning(
+            beregningId = beregningResultat.id,
+            behandlingId = behandlingId,
+            beregningsperioder = beregningResultat.beregningsperioder,
+            beregnetDato = beregningResultat.beregnetDato,
+            grunnlagMetadata = Metadata(
+                sakId = 0,
+                versjon = beregningResultat.grunnlagVersjon
+            )
+        )
+        val lagretBeregning = beregningRepository.lagre(beregningLagret)
+
+        val beregningHentet = beregningRepository.hent(behandlingId)
+
+        assertTrue(lagretBeregning.toJson() == beregningHentet.toJson())
```

Comment:
Kanskje bedre å sammenligne som en type `JsonNode` i stedet for?

Reply:
@Watercolours Noen tanker?

Reply:
Så vidt jeg ser er vel alt her `data class`, så kunne like så godt sammenlignet de uten å konvertere. 

Med det sagt ser jeg nå at @simejo må bytte ut `assertTrue` med `assertEquals`.